src/NServiceBus.Core.Tests/Transports/PushRuntimeSettingsTests.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.Core.Tests.Transports
+{
+    using System;
+    using NUnit.Framework;
+    using Transport;
+
+    [TestFixture]
+    public class PushRuntimeSettingsTests
+    {
+        [Test]
+        public void Should_default_concurrency_to_num_processors()
+        {
+            Assert.AreEqual(Math.Max(2, Environment.ProcessorCount), new PushRuntimeSettings().MaxConcurrency);
```

Comment:
This is a strange test. You are essentially duplicating the logic from the impl. For a "proper" test you would need to fake away the processor count.